src/frontend/src/catalog/schema_catalog.rs

```diff
@@ -192,7 +192,7 @@ impl SchemaCatalog {
                     .unwrap();
                 entry.get_mut().remove(pos);
             }
-            Vacant(_entry) => unreachable!(),
```

Comment:
Is this modification unintentional? I didn't see any reason to ignore it.

Reply:
Yes, it is intentional.
The notification order received by frontend is:
1. update table
  * create table in the new schema
  * drop table in the old schema (which calls `indexes_by_table_id.remove(table_id)`)
2. update index
  * create index in the new schema
  * drop index in the old schema (which lookup `indexes_by_table_id` in the old schema and it is removed in step 1)

By changing the `unreachable!()` to `()` and with the designate notification order we can reuse most of code like `drop_table/create_table/...`. Otherwise we may need to create a new kind of notification for `alter set schema` and reimplement all the drop and create logic. It might be more robust though since not sharing functions with other statements, but may introduce more duplicate and trivial code.

Reply:
I see. Well, let's just ignore it for now and add some comments for it.